app/forms/work_form.rb

```diff
@@ -2,8 +2,7 @@
 
 # Form for a Work
 class WorkForm < ApplicationForm
-  accepts_nested_attributes_for :related_links, :related_works
-  accepts_nested_attributes_for :publication_date, form_class: DateForm, array: false
+  accepts_nested_attributes_for :related_links, :related_works, :publication_date
```

Comment:
We can handle publication_date correctly after two changes:

1. Subclassing `DateForm` as `PublicationDateForm`, which obviates the need to specify a form class; and
2. Sniffing the nested attr to determine if it's plural or singular, and deducing from that whether the attr is repeatable or not.